### ISSUE TITLE
NonClient content using RenderingSurface seems to be clipped to Client area

### ISSUE THREAD

**Issue by Ident** [cegui/cegui#874]
**[Original report](https://bitbucket.org/cegui/cegui/issue/874) by Paul Turner (Bitbucket: [pturner](https://bitbucket.org/pturner), GitHub: [pturner](https://github.com/pturner)).**

----------------------------------------

http://www.cegui.org.uk/phpBB2/viewtopic.php?f=10&t=6168

The report is about the non rectangular hit testing wiki article, and modifying that to affect Titlebar rather than PushButton - but this merely sets up the required scenario to trigger the issue.

Basically it appears that if a NonClient window is using ARS, when it is rendered back to the parent surface it is clipped to the parent window client area and not the non-client area.

**Reproducibility:** always
